src/DeviceIdWidget/widget/DeviceIdWidget.ts

```diff
@@ -0,0 +1,133 @@
+import * as dojoDeclare from "dojo/_base/declare";
+import * as domConstruct from "dojo/dom-construct";
+import * as WidgetBase from "mxui/widget/_WidgetBase";
+
+type OnGetDeviceInformationOptions = "doNothing" | "showPage" | "callMicroflow";
+type onGetDeviceInformationFallback = "Android" | "IOS" | "Web";
+
+class DeviceIdWidget extends WidgetBase {
+    private deviceEntity: string;
+    private deviceIDAttribute: string;
+    private deviceTypeAttribute: string;
+    private onGetDeviceInformationMicroflow: string;
+    private onGetDeviceInformationOption: OnGetDeviceInformationOptions;
+    private onGetDeviceInformationFallback: onGetDeviceInformationFallback;
+    private onGetDeviceInformationPage?: string;
+    private tokenAttribute: string;
+
+    update(mxObject: mendix.lib.MxObject, callback: () => void) {
+        const warnings = this.validateProps();
+        if (warnings) {
+            domConstruct.create("div", {
+                class: "alert alert-danger",
+                innerHTML: warnings
+            }, this.domNode);
+        } else {
+            this.setUpWidgetDom();
+            let type = "";
```

Comment:
line 27 to 35 can be written as.
const type = window.device? window.device.platform : this.onGetDeviceInformationFallback;
const id = window.device? window.device.uuid : "";